### PR TITLE
Modify the logic for making copies of estimators using the scikit-learn `[get/set]_params` methods

### PR DESCRIPTION
**Pull request by hannanabdul55**
## What's there in this PR? 
- Modify the way in which we store copies of estimators. 
- Logic: 
  - If the estimator is a sklearn based `estimator`, get the parameters using `get_params` and store the estimator class. 
  - Else, store a pickled version of the estimator (current logic) and use that for creating copies.

### CODE REVIEW

File: fairlearn/reductions/_exponentiated_gradient/_lagrangian.py [hunk 1]
```diff
@@ -165,7 +174,13 @@ def best_h(self, lambda_vec):
         the vector of Lagrange multipliers `lambda_vec`.
         """
         classifier = self._call_oracle(lambda_vec)
-        def h(X): return classifier.predict(X)
+
+        def h(X):
+            pred = classifier.predict(X)
+            if getattr(pred, "flatten", None) is not None:
```
Comment (romanlutz) [inline fairlearn/reductions/_exponentiated_gradient/_lagrangian.py:178]: what's this for? Can you add a comment in the code?

**Review by romanlutz — COMMENTED** [review]
This looks good, but why not write the PR to merge into fairlearn?

**Comment by hannanabdul55** [comment]
I thought you'd like a review here first. Sorry, I will raise a PR in the fairlearn 👍

**Comment by hannanabdul55** [comment]
@romanlutz should I remove the notebook as part of the PR on `fairlearn` or should I include that too?

**Pull request closed without merging**